lib/cadet_web/controllers/user_controller.ex

```diff
@@ -22,20 +23,76 @@ defmodule CadetWeb.UserController do
       grade: grade,
       max_grade: max_grade,
       story: story,
-      xp: xp
+      xp: xp,
+      game_states: game_states
     )
   end
 
+  def update_game_states(conn, %{"gameStates" => new_game_states}) do
+    user = conn.assigns[:current_user]
+
+    case update(user, new_game_states) do
+      {:ok, nil} ->
+        text(conn, "OK")
+
+      {:error, {status, message}} ->
+        conn
+        |> put_status(status)
+        |> text(message)
+    end
+  end
+
+  def clear_up_game_states(conn, _) do
+    user = conn.assigns[:current_user]
+
+    case clear(user) do
+      {:ok, nil} ->
+        text(conn, "OK")
+
+      {:error, {status, message}} ->
+        conn
+        |> put_status(status)
+        |> text(message)
+    end
+  end
+
   swagger_path :index do
     get("/user")
-
     summary("Get the name and role of a user")
-
     security([%{JWT: []}])
+    produces("application/json")
+    response(200, "OK", Schema.ref(:UserInfo))
+    response(401, "Unauthorised")
+  end
 
+
+  swagger_path :update_game_states do
+    put("/user/game_states/save")
+    summary("update user's game states")
+    security([%{JWT: []}])
+    consumes("application/json")
     produces("application/json")
```

Comment:
This line can be removed since your `update_game_states` action only returns a text response